src/AppInstallerCLICore/Workflows/UninstallFlow.cpp

```diff
@@ -0,0 +1,201 @@
+            BOOL success = CreateProcessW(app, commandLine, NULL, NULL, FALSE, 0, NULL, NULL, &startupInfo, &processInfo);
+
+            CoTaskMemFree(app);
+            CoTaskMemFree(commandLine);
+
+            if (!success)
+            {
+                return GetLastError();
+            }
+
+            CloseHandle(processInfo.hThread);
+            wil::unique_process_handle process{ processInfo.hProcess };
+
+            // Wait for uninstallation to finish
+            while (!progress.IsCancelled())
+            {
+                DWORD waitResult = WaitForSingleObject(process.get(), 250);
+                if (waitResult == WAIT_OBJECT_0)
+                {
+                    break;
+                }
+
+                if (waitResult != WAIT_TIMEOUT)
+                {
+                    THROW_LAST_ERROR_MSG("Unexpected WaitForSingleObjectResult: %d", waitResult);
+                }
+            }
+
+            if (progress.IsCancelled())
+            {
+                return {};
+            }
+            else
+            {
+                DWORD exitCode = 0;
+                GetExitCodeProcess(process.get(), &exitCode);
+                return exitCode;
+            }
+        }
+    }
+
+    void GetUninstallInfo(Execution::Context& context)
+    {
+        auto installedPackageVersion = context.Get<Execution::Data::InstalledPackageVersion>();
+        const std::string installedTypeString = installedPackageVersion->GetMetadata()[PackageVersionMetadata::InstalledType];
+        switch (ManifestInstaller::ConvertToInstallerTypeEnum(installedTypeString))
+        {
+        case ManifestInstaller::InstallerTypeEnum::Exe:
+        case ManifestInstaller::InstallerTypeEnum::Burn:
+        case ManifestInstaller::InstallerTypeEnum::Inno:
+        case ManifestInstaller::InstallerTypeEnum::Msi:
+        case ManifestInstaller::InstallerTypeEnum::Nullsoft:
+        case ManifestInstaller::InstallerTypeEnum::Wix:
+        {
+            IPackageVersion::Metadata packageMetadata = installedPackageVersion->GetMetadata();
+            IPackageVersion::Metadata::iterator itr;
+
+            if (context.Args.Contains(Execution::Args::Type::Interactive))
+            {
+                itr = packageMetadata.find(PackageVersionMetadata::StandardUninstallCommand);
+            }
+            else if (context.Args.Contains(Execution::Args::Type::Silent))
+            {
+                itr = packageMetadata.find(PackageVersionMetadata::SilentUninstallCommand);
+            }
+            else
+            {
+                itr = packageMetadata.find(PackageVersionMetadata::StandardUninstallCommand);
+                if (itr == packageMetadata.end())
+                {
+                    itr = packageMetadata.find(PackageVersionMetadata::SilentUninstallCommand);
+                }
+            }
+
+            if (itr == packageMetadata.end())
+            {
+                AICLI_TERMINATE_CONTEXT(E_ABORT);
+            }
```

Comment:
There needs to be some `context.Reporter` output here to indicate that no uninstall string was available.  Also, you cannot use E_ABORT, as that is the HR used for user cancellation.  It is probably best to create a new HR in our list.